pydantic/_internal/_generate_schema.py

```diff
@@ -702,8 +702,6 @@ def _generate_dc_field_schema(
         )
 
     def _common_field_schema(self, name: str, field_info: FieldInfo, decorators: DecoratorInfos) -> _CommonField:
-        assert field_info.annotation is not None, 'field_info.annotation should not be None when generating a schema'
```

Comment:
I thought this assertion was safe and that we couldn't hit this branch, but apparently we need to handle this case. Easy enough fix.

Reply:
Would that be an unreasonable breaking change? It's to fix a bug.